### PR TITLE
Ionesio's codebase cleanup 2

### PR DESCRIPTION
**Pull request by IonesioJunior**
## Description
Please include a summary of the change, the motivation, and any additional context that will help others understand your PR. If it closes one or more open issues, [please tag them as described here](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

File: packages/syft/src/syft/lib/python/slice.py [hunk 1]
```diff
@@ -26,6 +32,15 @@ def __init__(
         step: Optional[Any] = None,
         id: Optional[UID] = None,
     ):
+        """
+        This class will receive start, stop, step and ID as valid parameters.
+
+        :param start: Index/position where the slicing of the object starts.
```
Comment (rasswanth-s) [inline packages/syft/src/syft/lib/python/slice.py:38]: MAYBE: could we switch to `Google Python` Docstring for consistency in the codebase.

File: packages/syft/src/syft/lib/python/slice.py [hunk 2]
```diff
@@ -47,52 +63,138 @@ def id(self) -> UID:
...
+        :return: returns a PySyft boolean checking if self > other.
+        :rtype: SyPrimitiveRet
+        """
+        res = self.value.__gt__(upcast(other))  # type: ignore
         return PrimitiveFactory.generate_primitive(value=res)
 
     def __le__(self, other: Any) -> SyPrimitiveRet:
-        res = self.value.__le__(other)  # type: ignore
+        """
+        Compare if self =< other.
```
Comment (rasswanth-s) [inline packages/syft/src/syft/lib/python/slice.py:100]: TYPO : `<=` in less than

File: packages/syft/src/syft/lib/python/slice.py [hunk 3]
```diff
@@ -47,52 +63,138 @@ def id(self) -> UID:
...
     def __eq__(self, other: Any) -> SyPrimitiveRet:
-        res = self.value.__eq__(other)
+        """
+        Compare if self == other.
+
+        :param Any other: Object to be compared.
+        :return: returns a PySyft boolean checking if self == other.
+        :rtype: SyPrimitiveRet
+        """
+        res = self.value.__eq__(upcast(other))
```
Comment (rasswanth-s) [inline packages/syft/src/syft/lib/python/slice.py:73]: good catch on the `upcast`  :+1:

**Review by rasswanth-s — APPROVED** [review]
Great work, nice documentation :rocket:

File: packages/syft/src/syft/lib/python/slice.py [hunk 4]
```diff
@@ -47,52 +63,138 @@ def id(self) -> UID:
         return self._id
 
     def __eq__(self, other: Any) -> SyPrimitiveRet:
-        res = self.value.__eq__(other)
+        """
+        Compare if self == other.
```
Comment (rasswanth-s) [inline packages/syft/src/syft/lib/python/slice.py:67]: MAYBE: shall we change to `google python` style docstring format?

File: packages/syft/tests/syft/lib/python/slice/slice_test.py [hunk 5]
```diff
@@ -124,6 +141,17 @@ def __eq__(self, other):
         self.assertEqual(s1, s1)
         self.assertRaises(Exc, lambda: s1.value == s2.value)
 
+    def test_id(self):
+        new_id = UID()
+        s1 = Slice(1, 2, 3, id=new_id)
+        assert new_id == s1.id
+
+    def test_upcast(self):
```
Comment (gmuraru) [inline packages/syft/tests/syft/lib/python/slice/slice_test.py:149]: Q: Should we also have a test when we test a `slice` against a `Slice`? (to see that the `__eq__` from slice works correctly?)

File: packages/syft/src/syft/lib/python/slice.py (hunk 4 above)
Comment (IonesioJunior) [reply packages/syft/src/syft/lib/python/slice.py:68]: Yup!

Comment (IonesioJunior) [reply packages/syft/src/syft/lib/python/slice.py:68]: Solved by fe2c81b

File: packages/syft/tests/syft/lib/python/slice/slice_test.py (hunk 5 above)
Comment (IonesioJunior) [reply packages/syft/tests/syft/lib/python/slice/slice_test.py:156]: Yeah, good point!

**Pull request merged**